app/websocket_app.rb

```diff
@@ -259,7 +259,7 @@ class WebsocketApp
   end
 
   def update_node(node_name, message)
-    node = FlightScheduler.app.nodes.fetch_or_add(node_name)
+    node = FlightScheduler.app.nodes.update_node(node_name)
```

Comment:
Figured you where going to say that :sweat_smile: 

During the refactor I found there was a corollary between  `update_nodes` and `fetch_or_add`. The `update_nodes` method won because it provides the super set of required features.

But I do agree, the naming could do work

Reply:
I've renamed `update_node` to be `register_node` as it best reflects the purposes of the method:
1. Ensure the node by the given `name` exists within the registry, and
2. Ensure the internal partition cache is up to date for the node.

There is no longer a `add` flag on this method as in practice the nodes _should_ always exist. The edge cases where they don't only occur in the specs. As such this is handled in `Node#attribute=` method pending refactoring.

https://github.com/openflighthpc/flight-scheduler-controller/pull/42/commits/3c27aa2273f7f8a69b8dede658045aa2cec098c2